core/src/main/scala/chisel3/internal/Binding.scala

```diff
@@ -176,3 +178,16 @@ private[chisel3] case class BundleLitBinding(litMap: Map[Data, LitArg]) extends
 private[chisel3] case class VecLitBinding(litMap: VectorMap[Data, LitArg]) extends LitBinding
 // Literal binding attached to a Property.
 private[chisel3] case object PropertyValueBinding extends UnconstrainedBinding with ReadOnlyBinding
+
+/** Binding for Property expressions.
+  *
+  * This binding is a little different, because Property expressions don't emit Nodes, so we sore all the information we
+  * need to create an in-memory Arg that captures the operation and operands.
+  *
+  * @param sourceInfo Source location where the expression was created
+  * @param enclosure BaseModule where the expression was created
+  * @param op Primitive operation for this expression
+  * @param args Arguments to this expression
+  */
+private[chisel3] case class PropExprBinding(sourceInfo: SourceInfo, enclosure: BaseModule, op: PropPrimOp, args: Arg*)
+    extends ConstrainedBinding
```

Comment:
So this is tricky and highlight the weirdness of us both having `chisel3.internal.ir._` and the binding. But basically binding traditionally was only used to store enough information to check if the lexical scoping of something--basically to check if it is legal to access it in a given scope. The normal IR (ie. the `_ref` of a `Data`) is where information for emitting FIRRTL was stored.

There was obviously a bit of feature creep in the Literal bindings and in the DataView bindings which include things that are arguably more appropriate for the ref than for the binding.

This is a mess, and we need to clean it up, but IMO both the PropPrimOp and the Args do not belong here and are in fact redundant with the information you're putting in the IR node `PropExpr`.

It's true that SourceInfo is also not usually stored here, but I think it is reasonable to do so since it's useful in error reporting.